doc/creating_commands.md

```diff
@@ -244,24 +218,80 @@ option "--attributes", "ATTRIBUTES", "Values of various attributes",
 
 ### Advanced option evaluation
 
-Sometimes it is necessary to tune the option values based on other parameters given on CLI.
+Sometimes it is necessary to tune or validate the option values based on other parameters given on CLI.
 An example could be setting default values based on other options, values lookup in a DB, etc.
-The right place for this are `OptionSources`. Abstract Hammer command uses two default option sources -
-`HammerCLI::Options::Sources::CommandLine` responsible for intial population of the options,
+The right place for this are option processors. There are two basic kinds of option processors in hammer:
+- *option sources* - they provide values for the options, descendants of `HammerCLI::Options::Sources::Base`
+- *option validators* - they check if the options values are valid, descendants of `HammerCLI::Options::Validators::Base`
+
+Option sources and validators can be mixed together. For example it's possible to collect options from the command line,
+do some validation on them and then continue with collecting options from a different source.
+The whole set of processors is invoked only once per command call. The processing is triggered by a first call
+to the `options` or `all_options` method, but at latest right after the option validation
+(before the command's `execute` method is invoked). The order is as follows:
+ 1. option parsing
+ 1. option normalization
+ 1. option processors execution (sources and validators)
+ 1. `execute` invocation
+
+#### Default option sources
+
+Abstract Hammer command uses two default option sources -
+`HammerCLI::Options::Sources::CommandLine` responsible for intial population of the options and
 `HammerCLI::Options::Sources::SavedDefaults` adding defaults managed by the `defaults` command.
```

Comment:
Could we add something like: The default option sources are wrapped in `DefaultInputs` source list so that the default hierarchy  of sources is
```
DefaultInputs
  CommandLine
  SavedDefaults
```